src/main/java/com/example/hop_oasis/service/data/CiderImageServiceImpl.java

```diff
@@ -22,44 +20,37 @@
 @RequiredArgsConstructor
 public class CiderImageServiceImpl implements CiderImageService {
     private final CiderImageRepository ciderImageRepository;
-    private final CiderImageMapper ciderImageMapper;
-    private final ImageCompressor imageCompressor;
     private final CiderRepository ciderRepository;
-
+    private final S3Service s3Service;
     @Override
-    public CiderImageDto getCiderImageByName(String name) {
-        Optional<CiderImage> imageOp = ciderImageRepository.findFirstByName(name);
-        if (imageOp.isEmpty()) {
-            throw new ResourceNotFoundException(RESOURCE_NOT_FOUND, name);
-        }
-        CiderImage image = imageOp.get();
-        image.setImage(imageCompressor.decompressImage(image.getImage(), name));
-        return ciderImageMapper.toDto(image);
+    public CiderImageUrlDto getCiderImageByName(String name) {
+     return new CiderImageUrlDto(s3Service.getFileUrl(name).toString());
     }
     @Override
-    public CiderImage addCiderImageToCider(Long ciderId, MultipartFile file) {
-        byte[] image;
+    public CiderImageUrlDto addCiderImageToCider(Long ciderId, MultipartFile file) {
         try {
-            image = imageCompressor.compressImage(file.getBytes());
+            String fileName ="ciders/" + file.getOriginalFilename();
+            s3Service.uploadFile(fileName, file);
         } catch (IOException e) {
             throw new ResourceNotFoundException(RESOURCE_NOT_FOUND, "");
         }
         CiderImage image1 = CiderImage.builder()
-                .image(image)
                 .name(file.getOriginalFilename())
                 .build();
         Cider cider = ciderRepository.findById(ciderId).orElseThrow(() ->
                 new ResourceNotFoundException(RESOURCE_NOT_FOUND, ciderId));
 
         image1.setCider(cider);
-       return ciderImageRepository.save(image1);
+        ciderImageRepository.save(image1);
```

Comment:
Why do you need separate repository for cider image? 
You can add field imageUrl to Cider and save it.
The same applies to beer image and product bundle image

Reply:
I will do this later